ambari-server/src/main/java/org/apache/ambari/server/configuration/Configuration.java

```diff
@@ -2851,6 +2854,41 @@ private static Properties readConfigFile() {
     return properties;
   }
 
+  /**
+   * Removing the given properties from ambari.properties (i.e. at upgrade time)
+   *
+   * @param propertiesToBeCleared
+   *          the properties to be removed
+   * @throws AmbariException
+   *           if there was any issue when clearing ambari.properties
+   */
+  public void removePropertiesFromAmbariProperties(Collection<String> propertiesToBeRemoved) throws AmbariException {
+    try {
+      final File ambariPropertiesFile = new File(Configuration.class.getClassLoader().getResource(Configuration.CONFIG_FILE).getPath());
+      final List<String> ambariPropertiesLines = FileUtils.readLines(ambariPropertiesFile, Charset.defaultCharset());
+      final StringBuilder newAmbariProperties = new StringBuilder();
+      for (String line : ambariPropertiesLines) {
+        if (!lineNeedsToBeRemoved(line, propertiesToBeRemoved)) {
+          newAmbariProperties.append(line).append(System.getProperty("line.separator"));
+        }
+      }
+      FileUtils.deleteQuietly(ambariPropertiesFile);
+      FileUtils.writeStringToFile(ambariPropertiesFile, newAmbariProperties.toString(), Charset.defaultCharset());
```

Comment:
Deleting the file before writing the new one may lead to data loss.

Reply:
Fixed